### ISSUE TITLE
project: support in obs queries

### ISSUE THREAD

**Issue by synrg** [dronefly-garden/dronefly#87]
Some thoughts about this:

- `[p]project add <abbrev> <id#>` (cf. `[p]place add <abbrev> <id#>`)
- `[p]obs <query> from <abbrev>` (or `[p]s obs <query> from <abbrev>`) to filter the matching observations from a particular project
- yes, we also use `from` for places, but this can be handled by:
    - first, lookup <abbrev> in places
    - second, lookup <abbrev> in projects
    - finally, as a fallback, if <abbrev> is a number, treat it as a place id#
    - we don't anticipate much overlap between the two for abbrevations, and if there are conflicts, we can support `from place <abbrev>` and `from project <abbrev>` to disambiguate. you'd need to use this form if you want to filter using a project id#, for example, as otherwise it will be treated as a place id# according to the above rules.
- using the above rules, the project abbrev should therefore be usable anywhere a place would have been used instead. that would extend also to using the :pencil: reaction to write in a place. you could put a project abbrev there instead and it would be added to the table, i.e. a project is considered a sort of "virtual place"

Michael and I discussed this polymorphism with respect to place/project in previous VC's and agree this should feel natural to users, so long as the headers/documentation are adjusted accordingly. Projects are "like a place", except they use arbitrary criteria for inclusion. That is not outside the expectations of anyone who is even only modestly computer literate.

An analogy from the average computer / smartphone user's experience is "virtual folders" like "Recent", which organize a collection of files recently accessed using the folder metaphor (a sort of "place" for files) even though they are actually stored in several different folders. Another virtual folder is "Pictures", particularly on smartphones, which aggregates files of the same type (image files) that belong to different apps & therefore are located in different actual folders on the phone.

Therefore, the idea that projects are a "sort of (virtual) place" should not be too hard to grasp, assuming any prior exposure to these concepts at all. This helps to keep the syntax for queries simple by default, with additional complexity (adding a keyword `place` or `project` sometimes) only being needed for uncommon corner cases.

**Comment by michaelpirrello** [comment]
![image](https://user-images.githubusercontent.com/63756824/84075716-9d797400-a9a2-11ea-9971-8da189ea919b.png)

**Comment by synrg** [comment]
The above illustrates two broken expectations:

- the first is that `from 2020` would retrieve fish for the iNaturalist Discord 2020 yearlisting project. it doesn't (yet; see the revised issue description above)
- the second is that `from 2020` would in any case match a result with the string "2020" in it. it doesn't. if there is no abbreviation match, the current rule (which looks at places only) will fall back to treating a number as a request to look up a record by project id#

We discussed this in our last VC. we feel people don't use id# much (few even know it exists) but it is important to retain for future bot functionality we would like to write, and also for "power user" current uses, such as curators accessing records that otherwise are inaccessible because Elasticsearch searches exclude the desired record(s) from the results.

**Comment by synrg** [comment]
I'm not convinced by my own projects are "like a place" argument. I have mulled over this for months, and each time I returned to this issue, I was uneasy with it. In the end, the operator overload problem will, I fear, cause a lot of confusion, not to mention require some additional code complexity to handle it.

To address this, since `in` is a more natural preposition for project inclusion, I propose we use another two word option here, `in prj`, following the `not by` and `id by` precedents.

I'm not sure if this should be supported at all for `,taxon` queries, though. It seems like more of an `,obs` or `,tab` filter. Enumerating all of the counts of a taxon per project added to table at the bottom of the embed labelled `obs# (spp#) in project:` with buttons to add/remove projects just doesn't seem very useful at all. Instead, if `in prj` is a part of an `,obs` or `,tab` query (or `,map` or `,map id`, recently added), then the *in project* criterion should be added to the topic of the embed.

Thus, `,tab ruffed grouse in prj feathers by me` would makes the topic of the embed: **Observations of *Bonasa umbellus* (Ruffed Grouse) in Found Feathers**, and the `by me` starts a `by user` table at the bottom with my contributions to the project in that taxon. Anyone else can press :hash: and add themselves to compare their contributions.
